sql/sql_class.h

```diff
@@ -2067,6 +2069,36 @@ class THD : public MDL_context_owner,
   ha_rows m_examined_row_count;
 
  private:
+  std::shared_ptr<explicit_snapshot> m_explicit_snapshot;
+
+ public:
+  /**
+   * Creates an explicit snapshot and associates it with the current conn
+   * return true if error, false otherwise
+   */
+  bool create_explicit_snapshot();
+
+  inline std::shared_ptr<explicit_snapshot> get_explicit_snapshot() {
+    return m_explicit_snapshot;
+  }
+
+  inline void set_explicit_snapshot(std::shared_ptr<explicit_snapshot> s) {
+    m_explicit_snapshot = s;
```

Comment:
use `std::move()` is shared_ptr is passed by value
```suggestion
    m_explicit_snapshot = std::move(s);
```